Utils.py

```diff
@@ -966,3 +967,13 @@ def __bool__(self):
 
     def __len__(self):
         return sum(len(iterable) for iterable in self.iterable)
+
+
+def is_iterable_of_str(obj: object) -> TypeGuard[typing.Iterable[str]]:
+    """ but not a `str` (because technically, `str` is `Iterable[str]`) """
+    if isinstance(obj, str):
+        return False
+    if not isinstance(obj, typing.Iterable):
```

Comment:
shouldn't this be `collections.abc.Iterable`?

Reply:
in Python 3.9+
`typing.Iterable` is an alias to `collections.abc.Iterable`
so they refer to the same object

In 3.8 it's a patched version of the same thing.
I tested it in a 3.8 repl

```python
>>> a = []
>>> import typing
>>> isinstance(a, typing.Iterable)
True
```

It will work with either one here.